File: converte_para_texto.cpp
#include <bits/stdc++.h>

using namespace std;


#define INPUTFILE "caterpillar_nim_1"
#define OUTPUTFILE "caterpillar_nim_1.txt"


int main() {
    fstream input_file;
    ofstream output_file;
    input_file.open(INPUTFILE, ios::in | ios::binary);
    output_file.open(OUTPUTFILE, ios::out | ios::trunc);

    unsigned int n;
    while (input_file.read((char*) &n, sizeof(unsigned int)))
        output_file << n << " ";
    output_file << endl;

    input_file.close();
    output_file.close();
}

File: pythonscripts/converte_para_texto.py
INPUTFILE = "data/caterpillar_nim_1"
OUTPUTFILE = "data/caterpillar_nim_1.txt"

with open(INPUTFILE, "rb") as input_file, open(OUTPUTFILE, "w") as output_file:
    while True:
        data = input_file.read(4)  # 4 bytes for an unsigned int
        if not data:
            break
        n = int.from_bytes(data, byteorder='little')
        output_file.write(f"{n} ")
    output_file.write("\n")

    print(f"Converted {INPUTFILE} to {OUTPUTFILE}")
    print(f"Number of bytes in file: {input_file.tell()}")
    print(f"Number of bytes read: {input_file.tell()}")
    print(f"Number of elements read: {input_file.tell() // 4}")
